server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RandomStoring.java

```diff
@@ -52,6 +54,9 @@
 
     private static final int MIN_NUMBER_OF_RECIPIENTS = 4;
     private static final int MAX_NUMBER_OF_RECIPIENTS = 8;
+    private static final Duration CACHE_DURATION = Duration.ofMinutes(15);
```

Comment:
do you think 15 minutes is long? Perhaps we will want 1 hours for storing mails in random mailboxes. That means only around 16 -> 32 mailboxes are used to store mails for a whole hour.

Reply:
In fact I wanted to avoid relaunching James during provisioning, 1 hours seems a bit long to me, at current rate it would imply that 51k to 101k mails would be in only a few user mailboxes (we dequeue ~211 messages per minute)

Reply:
okay

Reply:
Why should we limit the cache duration at all?

Reply:
To avoid being forced to restart james if we add mailboxes